File: codechef/ENFE2019/Hex_Prime_HXPRIM.py
LIMIT = int(10e5) + 6

def seive_generation():
    '''
    prime number generation process
    Generate a counter upto LIMIT that will store number of
    primes number that itself is prime and adding 6 will also
    be a prime
    '''
    seive = [False, False] + [True] * LIMIT
    ENDS = int(LIMIT ** 0.5)
    for i in range(2, ENDS):
        if seive[i]:
            j = i * i
            while j < LIMIT:
                seive[j] = False
                j += i

    #create a counter for hexprimes
    counter = [0] * LIMIT
    counts  =  0
    for i in range(2, LIMIT - 6):
        if seive[i] and seive[i+6]:counts += 1
        counter[i] = counts
    return counter

counts = seive_generation()
nb_test = int(input())
for _ in range(nb_test):
    current = int(input())
    print(counts[current])

File: codechef/ENFE2019/Hex_Prime_HXPRIM.go
package main

import "fmt"

var limit = 100006

func primes_counts() []int {
	limit = limit + 1
	primes := make([]bool, limit)
	// implies up to the sqrt of limit
	for k := 2; k*k <= limit; k++ {
		if primes[k] != true {
			for ix := k * k; ix < limit; ix += k {
				primes[ix] = true
			}
		}
	}

	count := 0
	counts := make([]int, limit)
	for ix := 2; ix < limit-6; ix++ {
		if primes[ix] != true && primes[ix+6] != true {
			count = count + 1
		}
		counts[ix] = count

	}
	return counts
}
func main() {
	var nbTest, current int
	counts := primes_counts()
	fmt.Scan(&nbTest)
	for nbTest > 0 {
		fmt.Scan(&current)
		fmt.Println(counts[current])
		nbTest--
	}

}
